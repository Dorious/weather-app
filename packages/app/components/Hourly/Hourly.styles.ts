import styled from 'styled-components';

export const Container = styled.section`
  border-top: 1px solid #fff;
  padding: 1rem 0;
  width: 100%;
  overflow: auto;
  white-space: nowrap;
  text-align: center;
`;

export const Day = styled.div`
  display: inline-block;
  width: 5rem;

  > img {
    width: 3rem;
    height: 3rem;
  }
`;

export const Hour = styled.div`
  margin-bottom: 2rem;
`;
