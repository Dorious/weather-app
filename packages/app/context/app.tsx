import { allowedStatusCodes } from 'next/dist/lib/load-custom-routes';
import { createContext, useContext, useState } from 'react';

export type AppState = {
  loading?: boolean,
  searchQuery?: string,
  setSearchQuery?: (query: string) => void; 
  geolocation?: GeolocationCoordinates;
  getGeolocation?: () => void;
};

export const initialState = {
  loading: false,
  searchQuery: '',
  setSearchQuery: () => {},
  geolocation: undefined,
  getGeolocation: () => {
    console.log('grr');
  }
};

const AppContext = createContext<AppState>(initialState);

export function AppWrapper({ children }) {
  const [searchQuery, setSearchQuery] = useState('');
  const [loading, setLoading] = useState(false);

  const appState = {
    ...initialState,
    loading,
    searchQuery,
    setSearchQuery,
  };

  return (
    <AppContext.Provider value={appState}>
      {children}
    </AppContext.Provider>
  );
}

export function useAppContext() {
  return useContext(AppContext);
}