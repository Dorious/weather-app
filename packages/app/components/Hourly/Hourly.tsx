import { memo } from 'react';
import { WeatherData } from '../../context/app.types';
import Symbol from '../Symbol';
import * as S from './Hourly.styles';

export type HourlyProps = {
  weatherData: WeatherData;
};

export default memo(function Hourly({ weatherData }: HourlyProps): JSX.Element {
  const timeseries = [...weatherData?.properties.timeseries].splice(0, 24);
  return (
    <S.Container>
      {timeseries.map((series, idx) => {
        const temperature = Math.round(
          series.data.instant.details.air_temperature
        );
        const code = series.data.next_1_hours.summary.symbol_code;

        return (
          <S.Day key={series.time}>
            <S.Hour>
              {idx === 0 ? 'Now' : new Date(series.time).getHours()}
            </S.Hour>
            <Symbol code={code}></Symbol>
            <div>{temperature}&deg;</div>
          </S.Day>
        );
      })}
    </S.Container>
  );
});
