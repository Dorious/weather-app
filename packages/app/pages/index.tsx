import { useAppContext } from '../context/app'
import Background from '../components/Background'
import CurrentWeather from '../components/CurrentWeather'
import SearchBar from '../components/SearchBar'

import * as S from './_app.styles'
import { useWeatherData } from '../hooks/useWeatherData'

export function Index(): JSX.Element {
  const { geolocation } = useAppContext()

  const errorMessage =
    geolocation &&
    geolocation instanceof GeolocationPositionError &&
    geolocation.message

  const { weatherData } = useWeatherData(geolocation)

  return (
    <>
      <S.AppContainer>
        <Background symbolCode="partialycloud_day" />
        <S.AppBody>
          <SearchBar disabled />
          {errorMessage ? <S.AppError>{errorMessage}</S.AppError> : null}
          <CurrentWeather
            code="partlycloudy_day"
            geolocation={geolocation}
            weatherData={weatherData}
          />
        </S.AppBody>
      </S.AppContainer>
    </>
  )
}

export default Index
