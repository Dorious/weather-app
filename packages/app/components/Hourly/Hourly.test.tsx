import { render } from '@testing-library/react';
import Hourly from '.';

import { WeatherData } from '../../context/app.types';

import weatherData from '../../__mocks__/weatherData.json';

describe('Hourly', () => {
  it('should render properly', () => {
    const { container } = render(
      <Hourly weatherData={weatherData.data as WeatherData} />
    );
    expect(container).toMatchSnapshot();
  });
});
