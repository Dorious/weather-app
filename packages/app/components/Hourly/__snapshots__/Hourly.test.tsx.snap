// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hourly should render properly 1`] = `
<div>
  <div
    class="Hourlystyles__Container-sc-9ewovh-0 jQdjbP"
  >
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        Now
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        7
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        11
      </div>
      <img
        alt="Light rain"
        src="/symbols/svg/lightrain.svg"
        title="Light rain"
      />
      <div>
        7
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        12
      </div>
      <img
        alt="Light rain"
        src="/symbols/svg/lightrain.svg"
        title="Light rain"
      />
      <div>
        7
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        13
      </div>
      <img
        alt="Rain"
        src="/symbols/svg/rain.svg"
        title="Rain"
      />
      <div>
        5
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        14
      </div>
      <img
        alt="Light rain"
        src="/symbols/svg/lightrain.svg"
        title="Light rain"
      />
      <div>
        5
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        15
      </div>
      <img
        alt="Rain"
        src="/symbols/svg/rain.svg"
        title="Rain"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        16
      </div>
      <img
        alt="Rain"
        src="/symbols/svg/rain.svg"
        title="Rain"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        17
      </div>
      <img
        alt="Light rain"
        src="/symbols/svg/lightrain.svg"
        title="Light rain"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        18
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        19
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        20
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        21
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        22
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        23
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        0
      </div>
      <img
        alt="Partly cloudy"
        src="/symbols/svg/partlycloudy_night.svg"
        title="Partly cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        1
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        2
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        3
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        3
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        4
      </div>
      <img
        alt="Partly cloudy"
        src="/symbols/svg/partlycloudy_night.svg"
        title="Partly cloudy"
      />
      <div>
        3
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        5
      </div>
      <img
        alt="Partly cloudy"
        src="/symbols/svg/partlycloudy_night.svg"
        title="Partly cloudy"
      />
      <div>
        3
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        6
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        2
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        7
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        3
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        8
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        4
        °
      </div>
    </div>
    <div
      class="Hourlystyles__Day-sc-9ewovh-1 jlJFrI"
    >
      <div
        class="Hourlystyles__Hour-sc-9ewovh-2 bbzqeG"
      >
        9
      </div>
      <img
        alt="Cloudy"
        src="/symbols/svg/cloudy.svg"
        title="Cloudy"
      />
      <div>
        5
        °
      </div>
    </div>
  </div>
</div>
`;
